torch/_dynamo/variables/builder.py

```diff
@@ -1115,6 +1116,13 @@ def wrap_tensor(self, value: torch.Tensor):
             )
         )
 
+        if (
+            isinstance(value, torch.Tensor)
+            and value.is_nested
+            and not isinstance(value, NestedTensor)
+        ):
+            unimplemented("torch.compile does not support strided NestedTensor")
```

Comment:
Today, NestedTensor when `layout=torch.jagged` are already able to be tracked as `TensorVariable`. We are adding these explicit errors for the `layout=torch.strided` case because that is no longer being actively worked on. Are you able to use `layout.jagged` for your case?

Reply:
Thank you @soulitzer 
Jagged nested tensor is compiled successfully and no break is seen. It goes well.
Will the strided nested tensor be deprecated in torch?

Reply:
> Will the strided nested tensor be deprecated in torch ?

Today, strided nested tensor supports certain use cases (e.g. more than one ragged dimensions) that jagged nested tensor does not cover, so it cannot be deprecated.

> BTW, I wonder why 2 nested tensors have different types. They are both created by nested tensor creation.

Yes, they have different types because they are implemented via different mechanisms. jagged is implemented as a wrapper tensor subclass which would show up as a different type (but still subclasses Tensor), but strided is implemented as cpp tensor subclass which still show up as tensor.

Reply:
Because strided nested tensor is show up as the native tensor in python level, does it mean it can be compiled by the dynamo like any other tensor? I suppose the dynamo cannot be aware of the tensor is strided nested or not. It would compile the strided nested tensor as normal tensor. Is my understanding right? Feel free to correct me.
Thank you

Reply:
> My intuition is that the torch.compile can treat nested tensor as normal tensor.

Yes, this is actually the route we are taking with jagged layout nested tensor! We wrote jagged layout as a python wrapper subclass so we could make use of the shared infrastructure in AOTAutograd to trace away the subclasses as you are suggesting. If we want to also compile NestedTensor, we would need to either do the desugaring some other way, or be able to handle subclasses within inductor.

Reply:
Thank you for reply.
The strided nested tensor is normal native tensor. We wonder why it cannot be compiled like normal tensor in torch.compile stack. Is it because the inductor cannot lower the strided nested tensor?
```
type(strided_nested)
<class 'torch.Tensor'>
```

Reply:
That's correct, there would need to be special handling there for the subclass

Reply:
Thank you @soulitzer 
May i know if there is any plan to support compile strided nested tensor in torch?

Reply:
BTW, I got the runtime error why compiling the jagged nested tensor creation method with torch.compile. Here is the error:
```
Traceback (most recent call last):
  File "/home/pt-gpu/zejun/nested/nested.py", line 39, in <module>
    c = compiled_func3(a, b)
  File "/home/pt-gpu/zejun/pytorch/torch/_dynamo/eval_frame.py", line 390, in _fn
    return fn(*args, **kwargs)
  File "/home/pt-gpu/zejun/pytorch/torch/_dynamo/convert_frame.py", line 945, in catch_errors
    return callback(frame, cache_entry, hooks, frame_state, skip=1)
  File "/home/pt-gpu/zejun/pytorch/torch/_dynamo/convert_frame.py", line 808, in _convert_frame
    result = inner_convert(
  File "/home/pt-gpu/zejun/pytorch/torch/_dynamo/convert_frame.py", line 400, in _convert_frame_assert
    return _compile(
  File "/home/pt-gpu/anaconda3/envs/zejun/lib/python3.9/contextlib.py", line 79, in inner
    return func(*args, **kwds)
  File "/home/pt-gpu/zejun/pytorch/torch/_dynamo/convert_frame.py", line 692, in _compile
    guarded_code = compile_inner(code, one_graph, hooks, transform)
  File "/home/pt-gpu/zejun/pytorch/torch/_dynamo/utils.py", line 265, in time_wrapper
    r = func(*args, **kwargs)
  File "/home/pt-gpu/zejun/pytorch/torch/_dynamo/convert_frame.py", line 545, in compile_inner
    out_code = transform_code_object(code, transform)
  File "/home/pt-gpu/zejun/pytorch/torch/_dynamo/bytecode_transformation.py", line 1036, in transform_code_object
    transformations(instructions, code_options)
  File "/home/pt-gpu/zejun/pytorch/torch/_dynamo/convert_frame.py", line 165, in _fn
    return fn(*args, **kwargs)
  File "/home/pt-gpu/zejun/pytorch/torch/_dynamo/convert_frame.py", line 507, in transform
    tracer.run()
  File "/home/pt-gpu/zejun/pytorch/torch/_dynamo/symbolic_convert.py", line 2138, in run
    super().run()
  File "/home/pt-gpu/zejun/pytorch/torch/_dynamo/symbolic_convert.py", line 844, in run
    while self.step():
  File "/home/pt-gpu/zejun/pytorch/torch/_dynamo/symbolic_convert.py", line 754, in step
    self.dispatch_table[inst.opcode](self, inst)
  File "/home/pt-gpu/zejun/pytorch/torch/_dynamo/symbolic_convert.py", line 2259, in RETURN_VALUE
    self.output.compile_subgraph(
  File "/home/pt-gpu/zejun/pytorch/torch/_dynamo/output_graph.py", line 944, in compile_subgraph
    self.compile_and_call_fx_graph(tx, list(reversed(stack_values)), root)
  File "/home/pt-gpu/anaconda3/envs/zejun/lib/python3.9/contextlib.py", line 79, in inner
    return func(*args, **kwds)
  File "/home/pt-gpu/zejun/pytorch/torch/_dynamo/output_graph.py", line 1143, in compile_and_call_fx_graph
    compiled_fn = self.call_user_compiler(gm)
  File "/home/pt-gpu/zejun/pytorch/torch/_dynamo/utils.py", line 265, in time_wrapper
    r = func(*args, **kwargs)
  File "/home/pt-gpu/zejun/pytorch/torch/_dynamo/output_graph.py", line 1218, in call_user_compiler
    raise BackendCompilerFailed(self.compiler_fn, e).with_traceback(
  File "/home/pt-gpu/zejun/pytorch/torch/_dynamo/output_graph.py", line 1199, in call_user_compiler
    compiled_fn = compiler_fn(gm, self.example_inputs())
  File "/home/pt-gpu/zejun/pytorch/torch/_dynamo/repro/after_dynamo.py", line 117, in debug_wrapper
    compiled_gm = compiler_fn(gm, example_inputs)
  File "/home/pt-gpu/zejun/pytorch/torch/__init__.py", line 1739, in __call__
    return compile_fx(model_, inputs_, config_patches=self.config)
  File "/home/pt-gpu/anaconda3/envs/zejun/lib/python3.9/contextlib.py", line 79, in inner
    return func(*args, **kwds)
  File "/home/pt-gpu/zejun/pytorch/torch/_inductor/compile_fx.py", line 1360, in compile_fx
    return aot_autograd(
  File "/home/pt-gpu/zejun/pytorch/torch/_dynamo/backends/common.py", line 58, in compiler_fn
    cg = aot_module_simplified(gm, example_inputs, **kwargs)
  File "/home/pt-gpu/zejun/pytorch/torch/_functorch/aot_autograd.py", line 903, in aot_module_simplified
    compiled_fn = create_aot_dispatcher_function(
  File "/home/pt-gpu/zejun/pytorch/torch/_dynamo/utils.py", line 265, in time_wrapper
    r = func(*args, **kwargs)
  File "/home/pt-gpu/zejun/pytorch/torch/_functorch/aot_autograd.py", line 533, in create_aot_dispatcher_function
    fw_metadata = run_functionalized_fw_and_collect_metadata(
  File "/home/pt-gpu/zejun/pytorch/torch/_functorch/_aot_autograd/collect_metadata_analysis.py", line 555, in inner
    dynamic_dims = {
  File "/home/pt-gpu/zejun/pytorch/torch/_functorch/_aot_autograd/collect_metadata_analysis.py", line 556, in <setcomp>
    i for i, s in enumerate(o.shape) if not is_concrete_int(s)
  File "/home/pt-gpu/zejun/pytorch/torch/fx/experimental/symbolic_shapes.py", line 176, in is_concrete_int
    if isinstance(a.node.expr, sympy.core.numbers.Integer):
torch._dynamo.exc.BackendCompilerFailed: backend='inductor' raised:
AttributeError: 'torch._C._SymNode' object has no attribute 'expr'

Set TORCH_LOGS="+dynamo" and TORCHDYNAMO_VERBOSE=1 for more information


You can suppress this exception and fall back to eager by setting:
    import torch._dynamo
    torch._dynamo.config.suppress_errors = True
```

Here is my case:
```
import torch

def func3(a, b):
    x = torch.nested.nested_tensor([a, b], layout=torch.jagged)
    x = x + 1.0
    x = x * 2.0
    x = x + 3.0
    x = torch.relu(x)
    return torch.abs(x)

a = torch.randn(3,3,device='cuda')
b = torch.randn(3,3,device='cuda')

compiled_func3 = torch.compile(func3)
c = compiled_func3(a, b)
print('nested value = ', nested.values())
print('c      value = ', c.values())
```